module/mempool/stdmap/backdata/mapBackData.go

```diff
@@ -1,145 +1,129 @@
-// - init: the function that initializes the entity when it is not found.
+// - key: The key for which the value should be adjusted.
+// - adjust: the function that adjusts the value.
+// - init: A function that initializes the value if the key is not present.
+//
 // Returns:
-//   - the entity.
-//   - a bool which indicates whether the entity was found (or created).
-func (b *MapBackData) GetWithInit(entityID flow.Identifier, init func() flow.Entity) (flow.Entity, bool) {
-	if b.Has(entityID) {
-		return b.ByID(entityID)
+// - the adjusted value.
+//
+// - a bool which indicates whether the value was adjusted.
+func (b *MapBackData[K, V]) AdjustWithInit(key K, adjust func(V) V, init func() V) (V, bool) {
+	if b.Has(key) {
+		return b.Adjust(key, adjust)
 	}
-	b.Add(entityID, init())
-	return b.ByID(entityID)
+	b.Add(key, init())
+	return b.Adjust(key, adjust)
 }
 
-// ByID returns the given entity from the backdata.
-func (b *MapBackData) ByID(entityID flow.Identifier) (flow.Entity, bool) {
-	entity, exists := b.entities[entityID]
-	if !exists {
-		return nil, false
+// Get returns the value for the given key.
+// Returns true if the key-value pair exists, and false otherwise.
+func (b *MapBackData[K, V]) Get(key K) (value V, ok bool) {
+	value, ok = b.dataMap[key]
+	if !ok {
+		return value, false
 	}
-	return entity, true
+	return value, true
 }
 
-// Size returns the size of the backdata, i.e., total number of stored (entityId, entity)
-func (b *MapBackData) Size() uint {
-	return uint(len(b.entities))
+// Size returns the number of stored key-value pairs.
+func (b *MapBackData[K, V]) Size() uint {
+	return uint(len(b.dataMap))
 }
 
-// All returns all entities stored in the backdata.
-func (b *MapBackData) All() map[flow.Identifier]flow.Entity {
-	entities := make(map[flow.Identifier]flow.Entity)
-	for entityID, entity := range b.entities {
-		entities[entityID] = entity
+// All returns all stored key-value pairs as a map.
+func (b *MapBackData[K, V]) All() map[K]V {
+	values := make(map[K]V)
+	for key, value := range b.dataMap {
+		values[key] = value
 	}
-	return entities
+	return values
 }
 
-// Identifiers returns the list of identifiers of entities stored in the backdata.
-func (b *MapBackData) Identifiers() flow.IdentifierList {
-	ids := make(flow.IdentifierList, len(b.entities))
+// Keys returns an unordered list of keys stored in the backdata.
+func (b *MapBackData[K, V]) Keys() []K {
+	keys := make([]K, len(b.dataMap))
 	i := 0
-	for entityID := range b.entities {
-		ids[i] = entityID
+	for key := range b.dataMap {
+		keys[i] = key
 		i++
 	}
-	return ids
+	return keys
 }
 
-// Entities returns the list of entities stored in the backdata.
-func (b *MapBackData) Entities() []flow.Entity {
-	entities := make([]flow.Entity, len(b.entities))
+// Values returns an unordered list of values stored in the backdata.
+func (b *MapBackData[K, V]) Values() []V {
+	values := make([]V, len(b.dataMap))
```

Comment:
```suggestion
	values := make([]V, 0, len(b.dataMap))
```

This will allow to get rid of `values[i] = value` and replace it with `values = append(values, value)`

Reply:
[88006a5](https://github.com/onflow/flow-go/pull/7098/commits/88006a547d8578cd235697786004d3b10529a0d8)